src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs

```diff
@@ -592,8 +589,7 @@ protected virtual bool IsAction(TypeInfo typeInfo, MethodInfo methodInfo)
             var routeTemplateProvider =
                 attributes
                 .OfType<IRouteTemplateProvider>()
-                .Where(a => !IsSilentRouteAttribute(a))
-                .SingleOrDefault();
+                .SingleOrDefault(a => !IsSilentRouteAttribute(a));
```

Comment:
Should we throw a nicer exception in that case? `SingleOrDefault`'s message isn't particularly intuitive.